src/providers/Password.ts

```diff
@@ -74,6 +74,18 @@ export interface PasswordConfig<DataModel extends GenericDataModel> {
   ) => WithoutSystemFields<DocumentByName<DataModel, "users">> & {
     email: string;
   };
+  /**
+   * Performs custom validation on password provided during sign up or reset.
+   *
+   * Otherwise the default validation is used (password is not empty and
+   * at least 8 characters in length).
+   *
+   * If the provided password is invalid, implementations must throw an Error.
+   *
+   * @param password the password supplied during "signUp" or
+   *                 "reset-verification" flows.
+   */
+  validate?: (password?: string) => void;
```

Comment:
Why is the password optional? Also thoughts on `validatePassword` or `validatePasswordRequirements`? (as opposed to validating the login / validating that the password is correct)

Reply:
Thanks, I like the longer/more clear name. I went with `validate` initially to match the other short API names - but clarity is good.

Also I made `password` optional b/c I guess I thought it might not be provided. But I guess in that case it can be modeled as empty string.